ls1intum/Artemis
# [Bugfix] Build artifact download

### Checklist

### Motivation and Context


See issue #2628. The download fails because basic authentication (name+password) is required, but we try to authenticate with a token.

### Description


- Use basic authentication for artifact download by adding a case to BambooAuthorizationInterceptor
- Display real filename with extension for download

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise
4. Navigate to the template build plan
![image](https://user-images.githubusercontent.com/16407766/103682121-ba9e1180-4f88-11eb-99c7-8feeb633030f.png)
5. Click on _Configure build plan_
![image](https://user-images.githubusercontent.com/16407766/103682201-d3a6c280-4f88-11eb-95d5-b59b8264777c.png)
6. Select the default job
![image](https://user-images.githubusercontent.com/16407766/103682292-fc2ebc80-4f88-11eb-90dd-eb6bfad56199.png)
7. Alter the goal of the maven task as seen in the screenshot
![image](https://user-images.githubusercontent.com/16407766/103682375-1b2d4e80-4f89-11eb-9b19-2361d5fc057c.png)
8. In the same view click on tab _Artifacts_
9. Create an artifact with the following settings
![image](https://user-images.githubusercontent.com/16407766/103682544-52036480-4f89-11eb-83f2-5a3a66e4ea3e.png)
10. Participate in the exercise and solve all test cases (copy the solution)
11. A download build result button should be shown. Click on it.
![image](https://user-images.githubusercontent.com/16407766/103682702-80813f80-4f89-11eb-94e0-56ac37b8dc5c.png)
--> Artemis should provide a download for the jar and display the fileName with the .jar extension

### Test Coverage



participation.service: 41% -> 45%
